scheduler/src/org/pentaho/platform/scheduler2/gc/RepositoryCleanerSystemListener.java

```diff
@@ -0,0 +1,222 @@
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2002-2015 Pentaho Corporation..  All rights reserved.
+ */
+
+package org.pentaho.platform.scheduler2.gc;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.api.engine.IPentahoSystemListener;
+import org.pentaho.platform.api.scheduler2.ComplexJobTrigger;
+import org.pentaho.platform.api.scheduler2.IJobFilter;
+import org.pentaho.platform.api.scheduler2.IScheduler;
+import org.pentaho.platform.api.scheduler2.Job;
+import org.pentaho.platform.api.scheduler2.JobTrigger;
+import org.pentaho.platform.api.scheduler2.SchedulerException;
+import org.pentaho.platform.api.scheduler2.SimpleJobTrigger;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.pentaho.platform.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This is a 5.4-only class. To use it, update <tt>systemListeners.xml</tt> by adding the following section:
+ * <pre>
+  &lt;bean id="repositoryCleanerSystemListener"
+        class="org.pentaho.platform.scheduler2.gc.RepositoryCleanerSystemListener"&gt;
+    &lt;property name="gcEnabled" value="true"/&gt;
+    &lt;property name="execute" value="now"/&gt;
+  &lt;/bean&gt;
+ * </pre>
+ * <tt>gcEnabled</tt> is a non-mandatory parameter, <tt>true</tt> by default. Use it to turn off the listener without
+ * removing its description from the XML-file
+ * <tt>execute</tt> is a parameter, that describes a time pattern of GC procedure. Supported values are:
+ * <ul>
+ *   <li><tt>now</tt> - for one time execution</li>
+ *   <li><tt>weekly</tt> - for every Monday execution</li>
+ *   <li><tt>monthly</tt> - for every first day of month execution</li>
+ * </ul>
+ * Note, that periodic executions will be planned to start at 0:00. If an execution was not started at that time,
+ * e.g. the server was shut down, then it will be started as soon as the scheduler is restored.
+ * @author Andrey Khayrutdinov
+ */
+public class RepositoryCleanerSystemListener implements IPentahoSystemListener, IJobFilter {
+
+  private final Log logger = LogFactory.getLog( RepositoryCleanerSystemListener.class );
+
+  enum Frequency {
+    NOW( "now" ) {
+      @Override public JobTrigger createTrigger() {
+        return new SimpleJobTrigger( new Date(),
+          new Date( Long.MAX_VALUE ), 0, 1 );
+      }
+    },
+
+    WEEKLY( "weekly" ) {
+      @Override public JobTrigger createTrigger() {
+        // execute each first day of week at 0 hours
+        return new ComplexJobTrigger( null, null, null, ComplexJobTrigger.MONDAY, 0 );
```

Comment:
Probably makes more sense to use ComplexJobTrigger.SUNDAY since it's the first day of the week (1) and it starts the work-week off clean.